src/view/domconverter.js

```diff
@@ -839,6 +839,50 @@ export default class DomConverter {
 	}
 
 	/**
+	 * Checks if given {Selection DOM Selection} boundaries are in correct places.
+	 *
+	 * Incorrect places for selection are:
+	 * * before or in the middle of inline filler sequence,
+	 * * inside DOM element that represents {@link module:engine/view/uielement~UIElement view ui element}.
+	 *
+	 * @param {Selection} domSelection DOM Selection object to be checked.
+	 * @returns {Boolean} `true` if given selection is at correct place, `false` otherwise.
+	 */
+	isCorrectDomSelection( domSelection ) {
+		return this._isCorrectDomSelectionPosition( domSelection.anchorNode, domSelection.anchorOffset ) &&
+			this._isCorrectDomSelectionPosition( domSelection.focusNode, domSelection.focusOffset );
+	}
+
+	/**
+	 * Checks if given DOM position is a correct place for selection boundary. See {@link ~isCorrectDomSelection}.
+	 *
+	 * @private
+	 * @param {Node} domParent Position parent.
```

Comment:
I don't think that you can have a selection in a DF and this method is used for selection boundaries so this should be `Element`.